### PR TITLE
add AirM2M_CORE_ESP32C3 board

### PR DESCRIPTION
**Pull request by dreamcmi**
Added support for the AIRM2M CORE-ESP32C3 board, which has already had a great impact and has excellent sales on Taobao in China (https://item.taobao.com/item.htm?spm=a1z10.1-c-s.w4004-24090382179.12.1ec655d0ILdwls&id=666974425430)
This pull request has been tested on CORE-ESP32C3

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -14258,3 +14258,111 @@ deneyapkart1A.menu.DebugLevel.verbose=Verbose
...
+AirM2M_CORE_ESP32C3.menu.CPUFreq.40=40MHz
+AirM2M_CORE_ESP32C3.menu.CPUFreq.40.build.f_cpu=40000000L
+AirM2M_CORE_ESP32C3.menu.CPUFreq.20=20MHz
+AirM2M_CORE_ESP32C3.menu.CPUFreq.20.build.f_cpu=20000000L
+AirM2M_CORE_ESP32C3.menu.CPUFreq.10=10MHz
+AirM2M_CORE_ESP32C3.menu.CPUFreq.10.build.f_cpu=10000000L
+
+AirM2M_CORE_ESP32C3.menu.FlashMode.dio=DIO
+AirM2M_CORE_ESP32C3.menu.FlashMode.dio.build.flash_mode=dio
+AirM2M_CORE_ESP32C3.menu.FlashMode.dio.build.boot=dio
```
Comment (me-no-dev) [inline boards.txt:14337]: If you have only one option, the menu item is not necessary

File: boards.txt [hunk 2]
```diff
@@ -14258,3 +14258,111 @@ deneyapkart1A.menu.DebugLevel.verbose=Verbose
...
+
+AirM2M_CORE_ESP32C3.menu.FlashFreq.80=80MHz
+AirM2M_CORE_ESP32C3.menu.FlashFreq.80.build.flash_freq=80m
+AirM2M_CORE_ESP32C3.menu.FlashFreq.40=40MHz
+AirM2M_CORE_ESP32C3.menu.FlashFreq.40.build.flash_freq=40m
+
+AirM2M_CORE_ESP32C3.menu.FlashSize.4M=4MB (32Mb)
+AirM2M_CORE_ESP32C3.menu.FlashSize.4M.build.flash_size=4MB
```
Comment (me-no-dev) [inline boards.txt:14345]: If you have only one option, the menu item is not necessary

File: boards.txt (hunk 1 above)
Comment (me-no-dev) [reply boards.txt:14337]: Does the board only support DIO? QIO is not an option?

File: boards.txt (hunk 2 above)
Comment (dreamcmi) [reply boards.txt]: I have fixed it, please check

File: boards.txt [hunk 3]
```diff
@@ -14258,3 +14258,104 @@ deneyapkart1A.menu.DebugLevel.verbose=Verbose
...
+AirM2M_CORE_ESP32C3.upload.wait_for_upload_port=false
+
+AirM2M_CORE_ESP32C3.serial.disableDTR=false
+AirM2M_CORE_ESP32C3.serial.disableRTS=false
+
+AirM2M_CORE_ESP32C3.build.tarch=riscv32
+AirM2M_CORE_ESP32C3.build.target=esp
+AirM2M_CORE_ESP32C3.build.mcu=esp32c3
+AirM2M_CORE_ESP32C3.build.core=esp32
+AirM2M_CORE_ESP32C3.build.variant=esp32c3
```
Comment (me-no-dev) [inline boards.txt:14281]: almost missed this. If you want to use the variant header that you added here, you need to change the variant to the folder name. In your case that is `AirM2M_CORE_ESP32C3`

Comment (dreamcmi) [reply boards.txt]: I have fixed it, please check

**Comment by me-no-dev** [comment]
merged :)

**Pull request merged**